### ISSUE TITLE
`exclude` option doesn't work in the vscode extension

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#31]
workaround: use `ignore` instead

whats even the difference between these 2 options? one of them should be deleted i think

**Comment by DetachHead** [comment]
> whats even the difference between these 2 options? one of them should be deleted i think

https://docs.basedpyright.com/#/configuration?id=main-configuration-options

> - **exclude** [array of paths, optional]: Paths of directories or files that should not be considered part of the project. These override the includes directories and files, allowing specific subdirectories to be excluded. Note that files in the exclude paths may still be included in the analysis if they are referenced (imported) by source files that are not excluded. 

> - **ignore** [array of paths, optional]: Paths of directories or files whose diagnostic output (errors and warnings) should be suppressed even if they are an included file or within the transitive closure of an included file. Paths may contain wildcard characters

**Comment by DetachHead** [comment]
i think `exclude` and `include` are only relevant for the vscode extension when `basedpyright.analysis.diagnosticMode` is set to `"workspace"`, because when it's set to `"openFilesOnly"`, those open files are always included. maybe the documentation just needs to be updated to make this clear

**Comment by DetachHead** [comment]
oh also if you only use `ignore` but not `exclude`, the cli wastes time analyzing the ignored files even though it's not going to show any of its diagnostics. `ignore` should just imply `exclude` as well.

actually come to think of it, what's even the use case for excluding files but not ignoring their errors? maybe i was right in my first post that both options aren't needed

**Comment by DetachHead** [comment]
another thing to consider is that in the language server when you want to ignore all errors for a file, you might still want to analyze the file so that other language server features still work (semantic highlighting, hover, etc)

**Comment by DetachHead** [comment]
from https://github.com/DetachHead/basedpyright/issues/1134#issuecomment-3328274199 - if a file is `ignore`d it probably makes sense to suppress not only the diagnostics, but any language server errors that occur when analyzing the file

**Comment by beauxq** [comment]
I think I see value in separate `exclude` and `ignore`.

---

- I would use `ignore` when I don't want to report type errors in the code but I want "find all references" and "rename" to work with it (the best it can without good typing).
  - For example: a tightly coupled plugin module with a different author where I can't convince that author to annotate types (a real world situation I have in mind).

- I would use `exclude` when I need to copy some 3rd party code in my project for some reason. I don't want the type-checker/langserver dealing with it at all, I just need it to be there.

---

So `ignore` should not imply `exclude`.

On the question of whether `exclude` should imply `ignore`, I'm not as sure.
If I decide to open the excluded file in my editor, I might still want reports of type errors, and maybe auto-complete, and semantic highlighting. But if I do "find all references" from non-excluded code, I think I probably still wouldn't want it to bother with the excluded file that I had opened...
I'm not sure - That situation is a little muddy, so I probably wouldn't complain with whatever it does.

**Comment by DetachHead** [comment]
i agree, although this distinction is only relevant in the language server. when running basedpyright with the CLI, i don't see a reason for the two separate options
